controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -637,287 +730,226 @@ private boolean verifyRecordMatchesInput(List<Long> segmentsToSeal, List<SimpleE
-                                    .thenCompose(v -> updateHistoryTable(updated))
-                                    .whenComplete((r, e) -> {
-                                        if (e == null) {
-                                            log.debug("{}/{} scale op for epoch {}. Creating new epoch and updating history table.",
-                                                    scope, name, activeEpoch);
-                                        } else {
-                                            log.warn("{}/{} scale op for epoch {}. Failed to add partial record to history table. {}",
-                                                    scope, name, activeEpoch, e.getClass().getName());
-                                        }
-                                    });
-                        }));
+    public CompletableFuture<VersionedMetadata<EpochTransitionRecord>> getEpochTransition() {
+        return getEpochTransitionNode()
+                .thenApply(x -> new VersionedMetadata<>(EpochTransitionRecord.fromBytes(x.getData()), x.getVersion()));
     }
 
     private CompletableFuture<Void> clearMarkers(final Set<Long> segments) {
         return Futures.toVoid(Futures.allOfWithResults(segments.stream().parallel()
-                .map(this::removeColdMarker).collect(Collectors.toList())));
+                                                               .map(this::removeColdMarker).collect(Collectors.toList())));
     }
-
-    /**
-     * Remainder of scale metadata update. Also set the state back to active.
-     * 4. complete entry into the history table.
-     * 5. Add entry into the index table.
-     * 6. Delete epoch transition record.
-     *
-     * @param sealedSegmentSizes sealed segments with sizes
-     * @param record
-     * @return : list of newly created segments
-     */
+    
     @Override
-    public CompletableFuture<Void> scaleOldSegmentsSealed(Map<Long, Long> sealedSegmentSizes, VersionedMetadata<EpochTransitionRecord> record) {
-        return Futures.toVoid(clearMarkers(record.getObject().getSegmentsToSeal())
-                                    .thenCompose(x -> {
-                                        EpochTransitionRecord epochTransition = record.getObject();
-                                        return completePartialRecordInHistory(sealedSegmentSizes, epochTransition.getActiveEpoch(), 
-                                                epochTransition.getNewEpoch(), epochTransition.getTime());
-                                    }));
+    public CompletableFuture<Void> scaleOldSegmentsSealed(Map<Long, Long> sealedSegmentSizes,
+                                                          VersionedMetadata<EpochTransitionRecord> record) {
+        EpochTransitionRecord epochTransition = record.getObject();
+        return Futures.toVoid(clearMarkers(epochTransition.getSegmentsToSeal())
+                .thenCompose(x -> updateSealedSegmentSizes(sealedSegmentSizes))
```

Comment:
Why do we need to update sizes of sealed segments when scaling?

Reply:
We record sizes for any segments that are sealed right after we seal them. 

And we seal segments in three cases:
1. scale
2. rolling txn
3. seal stream